### ISSUE TITLE
Connection between Leafnode and Core NATS over satellite link fails to get established

### ISSUE THREAD

**Issue by piotrgalecki** [nats-io/nats-server#5417]
### Observed behavior

Siden is running into an odd issue with NATS Leafnode not being able to connect to our Core NATS server.
After TCP 3-way handshake Core NATS sends INFO message and then NATS Leafnode responds with INFO and 400+ms later sends closes connection (sends TCP FIN).
The latency of connection is 500ms.

Here are packets exchanged. Packets with P flag are INFO messages, followed by FIN packet
```
22:00:32.440826 IP 10.99.0.99.34786 > 3.136.10.169.7422: Flags [S], seq 1607668652, win 65535, options [mss 1460,sackOK,TS val 2932985413 ecr 0,nop,wscale 8], length 0
22:00:32.980351 IP 3.136.10.169.7422 > 10.99.0.99.34786: Flags [S.], seq 2383013068, ack 1607668653, win 65535, options [mss 1460,sackOK,TS val 1779976004 ecr 2932985413,nop,wscale 8], length 0
22:00:32.980410 IP 10.99.0.99.34786 > 3.136.10.169.7422: Flags [.], ack 1, win 256, options [nop,nop,TS val 2932985953 ecr 1779976004], length 0
22:00:33.520223 IP 3.136.10.169.7422 > 10.99.0.99.34786: Flags [P.], seq 1:468, ack 1, win 256, options [nop,nop,TS val 1779976544 ecr 2932985953], length 467
22:00:33.520276 IP 10.99.0.99.34786 > 3.136.10.169.7422: Flags [.], ack 468, win 261, options [nop,nop,TS val 2932986493 ecr 1779976544], length 0
22:00:33.520719 IP 10.99.0.99.34786 > 3.136.10.169.7422: Flags [P.], seq 1:164, ack 468, win 261, options [nop,nop,TS val 2932986493 ecr 1779976544], length 163
22:00:33.981008 IP 10.99.0.99.34786 > 3.136.10.169.7422: Flags [F.], seq 164, ack 468, win 261, options [nop,nop,TS val 2932986954 ecr 1779976544], length 0
```
in the console the NATS Leafnode outputs the following:
[7] [INF] 3.13.54.170:7422 - lid:53 - Leafnode connection closed: Read Error - Account: $G
[7] [INF] 3.13.54.170:7422 - lid:54 - Leafnode connection created for account: $G
[7] [INF] 3.13.54.170:7422 - lid:54 - Leafnode connection closed: Read Error - Account: $G

After investigation the issue is caused by too short timeout DEFAULT_LEAFNODE_INFO_WAIT, which is hardcoded to 1s.
Unfortunately it's a constant and not configurable,
We changed the timeout value in the vendor directory to 3s and that resolved the issue.

Please make this parameter configurable.

### Expected behavior

NATS connection can be established successfully

### Server and client version

v2.10.10

### Steps to reproduce

Test NATS Leafnode - Server connection over high latency link.

**Comment by yeongjonglim** [comment]
Hi @piotrgalecki, we are facing similar issue with the server version 2.10.24.

Is there any chance that the workaround you mentioned can be shared with more details on how that constant can be manually changed until the parameter is released?

**Comment by kozlovic** [comment]
@wallyqs @bruth @neilalexander I was surprised that @yeongjonglim was saying that the new option was not released based on the age of the PR in which I had made the changes.

The PR was not marked as a 2.11.0 (although a new option was added, so I could see why it could have been), and was first cherry-picked to be in 2.10.15 (see https://github.com/nats-io/nats-server/pull/5430), but that cherry-pick PR was closed in favor of (https://github.com/nats-io/nats-server/pull/5429) that did not include the original PR (https://github.com/nats-io/nats-server/pull/5424). It seems that there was no release 2.10.15, but I checked 2.10.16 and the latest and none contain the changes of this PR.

Again, if it is intentional that is not included in any release prior to 2.11.0, then all is good. If not, then that PR failed through the crack and you may want to include it in the upcoming 2.10.26.

**Comment by derekcollison** [comment]
Let's revisit and see if we should pull in. @neilalexander can you take the lead here?

**Comment by neilalexander** [comment]
Yep, at the time we suggested 2.11 because it was adding a new configuration option (and we didn't think 2.11 would take so long).

That said it's a safe enough change and it picks in cleanly still, not to mention that we've also added another option for the stall gates anyway, so I've added it for inclusion into 2.10.26-RC.5.

**Comment by piotrgalecki** [comment]
Will the fix be included in 2.11?
Just want to confirm.

**Comment by neilalexander** [comment]
Yes, both 2.10.26 and 2.11.
